src/thresholds.service.js

```diff
@@ -55,19 +60,28 @@ class ThresholdsService {
       weeksOfStock = plan.weeksOfStock
     }
 
-    let thresholds = Object.keys(weeklyLevels).reduce((index, product) => {
-      index[product] = Object.keys(weeksOfStock).reduce((productThresholds, threshold) => {
-        productThresholds[threshold] = Math.round(weeklyLevels[product] * weeksOfStock[threshold])
+    let thresholds = Object.keys(weeklyLevels).reduce((index, productId) => {
+      index[productId] = Object.keys(weeksOfStock).reduce((productThresholds, threshold) => {
+        const level = weeklyLevels[productId] * weeksOfStock[threshold]
+        const product = find(products, isId.bind(null, productId))
+        if (!product || !product.presentation) {
+          return productThresholds
```

Comment:
possibly. If the passed in product list doesn't contain what is expected probably throwing would be better, if we're going to ticket it separately, i would prefer to have it return unrounded levels as that is the least disruptive change. wdyt?